tools/pyi/gen_pyi.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import print_function
+import os
+import sys
+import inspect
+
+needed_modules = set()
+
+
+def type_to_python(typename):
+    typename = {'int64_t': 'int',
+                'Scalar': 'Union[float, int]',
+                'ScalarType': 'dtype',
+                'Device': 'Union[device, str, None]',
+                'IntList': 'Union[Tuple[int, ...], List[int, ...]]',
+                'Tensor?': 'Optional[Tensor]',
+                'Tensor': 'Tensor',
+                'bool': 'bool',
+                'double': 'float',
+                'Generator *': 'Generator',
+                'Generator*': 'Generator',
+                'std::vector<Tensor>': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'TensorList': 'Union[Tuple[Tensor, ...],List[Tensor, ...]]',
+                'Storage': 'Storage',  # inaccurate (FloatStorage...)
+                'SparseTensorRef': 'Tensor',
+                'void': 'None',
+                'Layout': 'layout',
+                'void*': 'int',  # dataptr
+                'std::string': 'str',
+                'real': 'Union[float, int]',
+                'accreal': 'Union[float, int]',
+                'IntegerTensor': 'Tensor',
+                'BoolTensor': 'Tensor',
+                'IndexTensor': 'Tensor',
+                }[typename]
+    # handle IntList[]
+    return typename
+
+
+def arg_to_type_hint(arg):
+    name = arg['name']
+    if name == 'from':  # keyword...
```

Comment:
lol. I wonder what our binding code does today. We should probably refuse to emit declarations that have keywords. This is a reasonable workaround for now.